docs/content/contributing/terraform/terraform-contribution-flow/_index.md

```diff
@@ -0,0 +1,316 @@
+
+<img src="/Azure-Verified-Modules/img/contribution/forkUpstream.png" alt="Upstream to fork and source repository." width=50%>
+
+{{< /hint >}}
+
+<br>
+
+---
+
+<br>
+
+<!--
+TODO:
+
+Checklist
+
+Run grept
+Update provider versions
+cd root
+avmfix -folder . # examples too
+make fmt
+make docs
+
+-->
+
+{{< hint type=note >}}
+
+Each time in the following sections we refer to 'your xzy', it is an indicator that you have to change something in your own environment.
+
+{{< /hint >}}
+
+## Prepare your developer environment
+
+### 1. Fork the module source repository
+
+{{< hint type=important >}}
+
+Each Terraform AVM module will have its own GitHub repository in the [`Azure`](https://github.com/Azure) GitHub Organization as per [SNFR19](/Azure-Verified-Modules/specs/shared/#id-snfr19---category-publishing---registries-targeted).
+
+This repository will be created by the Module owners and the AVM Core team collaboratively, including the configuration of permissions as per [SNFR9](/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions)
+
+{{< /hint >}}
+
+Module contributors are expected to fork the corresponding repository and work on a branch from within their fork, before then creating a Pull Request (PR) back into the source repository's `main` branch.
+
+To do so, simply navigate to your desired repository, select the `'Fork'` button to the top right of the UI, select where the fork should be created (i.e., the owning organization) and finally click 'Create fork'.
+
+{{< hint type=note >}}
+
+If the module repository you want to contribute to is not yet available, please get in touch with the respective module owner which can be tracked in the [Terraform Resource Modules index](https://azure.github.io/Azure-Verified-Modules/indexes/terraform/tf-resource-modules/) see `PrimaryModuleOwnerGHHandle` column.
+
+{{< /hint >}}
+
+<br>
+
+---
+
+<br>
+
+### 2. Setup your GitHub repository
+
+1. Set up a GitHub repository environment called `test`.
+<!-- TODO: secrets can be removed since the latest azteraform docker image with having ./avm implemented -->
+2. Create the following environment secrets in the `test` environment
+
+- `AZURE_CLIENT_ID`
+- `AZURE_TENANT_ID`
+- `AZURE_SUBSCRIPTION_ID`
+
+1. **Optional**: Create deployment protection rules for the `test` environment to avoid spinning up e2e tests with every pull request raised by third-parties.
+
+{{< hint type=note >}}
+
+There is a move from the `test` environment to GitHub self-hosted runners (1ES Pool) which soon makes the creation of secrets obsolete. This is currently in progress.
+
+{{< /hint >}}
+
+<br>
+
+---
+
+<br>
+
+### 3. Setup your Azure test environment
+
+AVM performs end-to-end (e2e) test dpeloyments of all modules in Azure for validation and requires an User-assigned Managed Identity (UAMI) with access to a test subscription:
```

Comment:
Need to revise this based on new runners - ping me so we can work it out.

Reply:
Marking this as `TBD` in our call.